clippy_lints/src/manual_let_else.rs

```diff
@@ -162,61 +163,107 @@ fn emit_manual_let_else(cx: &LateContext<'_>, span: Span, expr: &Expr<'_>, pat:
     );
 }
 
-fn expr_diverges(cx: &LateContext<'_>, expr: &'_ Expr<'_>) -> bool {
-    fn is_never(cx: &LateContext<'_>, expr: &'_ Expr<'_>) -> bool {
-        if let Some(ty) = cx.typeck_results().expr_ty_opt(expr) {
-            return ty.is_never();
-        }
-        false
+/// Check whether an expression is divergent. May give false negatives.
+pub fn expr_diverges(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
+    enum Descend {
+        Yes,
+        No,
```

Comment:
👋

Does the `Descend` from `clippy_utils` not work here?

Reply:
Right, it is `pub`, pushing a fix.